src/main/java/com/arpnetworking/metrics/mad/parsers/CollectdJsonToRecordParser.java

```diff
@@ -0,0 +1,389 @@
+        OBJECT_MAPPER.registerModule(queryLogParserModule);
+    }
+
+    /**
+     * Represents one record in a Collectd post body.
+     */
+    @Loggable
+    public static final class CollectdRecord {
+        public String getHost() {
+            return _host;
+        }
+
+        public DateTime getTime() {
+            return _time;
+        }
+
+        public String getPlugin() {
+            return _plugin;
+        }
+
+        public String getPluginInstance() {
+            return _pluginInstance;
+        }
+
+        public String getType() {
+            return _type;
+        }
+
+        public String getTypeInstance() {
+            return _typeInstance;
+        }
+
+        public List<Sample> getSamples() {
+            return _samples;
+        }
+
+        private CollectdRecord(final Builder builder) {
+            _host = builder._host;
+            _time = new DateTime(Math.round(builder._time * 1000));
+            _plugin = builder._plugin;
+            _pluginInstance = builder._pluginInstance;
+            _type = builder._type;
+            _typeInstance = builder._typeInstance;
+            _samples = Lists.newArrayListWithExpectedSize(builder._values.size());
+
+            final Iterator<Double> valuesIterator = builder._values.iterator();
+            final Iterator<String> typesIterator = builder._dsTypes.iterator();
+            final Iterator<String> namesIterator = builder._dsNames.iterator();
+            while (valuesIterator.hasNext() && typesIterator.hasNext() && namesIterator.hasNext()) {
+                _samples.add(new Sample(valuesIterator.next(), typesIterator.next(), namesIterator.next()));
+            }
+        }
+
+        private final String _host;
+        private final DateTime _time;
+        private final String _plugin;
+        private final String _pluginInstance;
+        private final String _type;
+        private final String _typeInstance;
+        private final List<Sample> _samples;
+
+        /**
+         * Represents a single sample in a collectd metric post.
+         */
+        public static final class Sample {
+            public double getValue() {
+                return _value;
+            }
+
+            public String getDsType() {
+                return _dsType;
+            }
+
+            public String getDsName() {
+                return _dsName;
+            }
+
+            /**
+             * Public constructor.
+             *
+             * @param value The value
+             * @param dsType The DS type
+             * @param dsName The DS name
+             */
+            public Sample(final double value, final String dsType, final String dsName) {
+                _value = value;
+                _dsType = dsType;
+                _dsName = dsName;
+            }
+
+            private final double _value;
+            private final String _dsType;
+            private final String _dsName;
+        }
+        /**
```

Comment:
Missing newline. I would prefer the builder before any other inner classes since those other inner classes will likely have builders and you lose a lot of context that way.